File: ppmtool/src/main/java/com/udemykanban/ppmtool/domain/ProjectTask.java
package com.udemykanban.ppmtool.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.util.Date;

@Entity
@Data @NoArgsConstructor
public class ProjectTask {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(updatable = false)
    private String projectSequence;

    @NotBlank(message = "Please include a project summary")
    private String summary;

    private String acceptanceCriteria;
    private String status;
    private Integer priority;
    private Date dueDate;

    //Many to One with Backlog
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.REFRESH)
    @JoinColumn(updatable = false, nullable = false)
    @JsonIgnore
    private Backlog backlog;

    @Column(updatable = false)
    private String projectIdentifier;


    private Date created_At;
    private Date updated_At;

    @PrePersist
    protected void onCreate(){
        this.created_At = new Date();
    }

    @PreUpdate
    protected void onUpdate(){
        this.updated_At = new Date();
    }

}

File: front-end-react/src/models/ProjectTask.ts
export class ProjectTask {
	id?: number;
	projectIdentifier: string;
	projectSequence: string;
	summary: string;
	acceptanceCriteria: string;
	status: string;
	priority: number;
	dueDate: string | undefined;

	constructor(
		$projectIdentifier: string,
		$projectsummary: string,
		$acceptanceCriteria: string,
		$status: string,
		$priority: number,
		$dueDate: string | undefined
	) {
		this.projectIdentifier = $projectIdentifier;
		this.projectSequence = "";
		this.summary = $projectsummary;
		this.acceptanceCriteria = $acceptanceCriteria;
		this.status = $status;
		this.priority = $priority;
		this.dueDate = $dueDate;
		this.id = undefined;
	}
}
